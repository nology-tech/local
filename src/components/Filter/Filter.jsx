import "./Filter.scss";
import Button from "../Button/Button";
import favArray from "../../data/favourites";
import { useState } from "react";

const Filter = () => {
  const [foodDrinkFilter, setFoodDrinkFilter] = useState(false);
  const [cultureFilter, setCultureFilter] = useState(false);
  const [otherFilter, setOtherFilter] = useState(false);

  const handleClickFoodDrink = (event) => {
    const userClick = event.target.value;
    setFoodDrinkFilter(userClick);
    console.log(userClick);
  };

  const handleClickCulture = (event) => {
    const userClick = event.target.value;
    setCultureFilter(userClick);
    console.log(userClick);
  };

  const handleClickOther = (event) => {
    const userClick = event.target.value;
    setOtherFilter(userClick);
    console.log(userClick);
  };

  const favFilter = favArray.filter((fav) => {
    if (foodDrinkFilter) {
      return fav.category.includes("Food & Drink");
    } else if (cultureFilter) {
      return fav.category.includes("Culture");
    } else if (otherFilter) {
      return fav.category.includes("Other");
    }
  });

  return (
    <div>
      <Button
        buttonText={"FOOD & DRINK"}
        buttonName="filter"
        onClick={handleClickFoodDrink}
      />
      <Button
        buttonText={"CULTURE"}
        buttonName="filter"
        onClick={handleClickCulture}
      />
      <Button
        buttonText={"OTHER"}
        buttonName="filter"
        onClick={handleClickOther}
      />
    </div>
  );
};

export default Filter;
