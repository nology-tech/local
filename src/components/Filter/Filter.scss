.filter {
  display: flex;
  overflow-x: scroll;
  padding-top: 1rem;
  padding-bottom: 2rem;
}
