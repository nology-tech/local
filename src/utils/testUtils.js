import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
