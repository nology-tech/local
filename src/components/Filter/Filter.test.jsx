import { customRender } from "../../utils/testUtils";
import { screen } from "@testing-library/react";
import Filter from "./Filter";
import userEvent from "@testing-library/user-event";

const mockData = ["FOOD & DRINK", "CULTURE", "OTHER"];

it("should render the filter component", () => {
  const mockFunction = jest.fn();
  const { container } = customRender(
    <Filter favArray={mockData} setallCardsArr={mockFunction} />
  );
  expect(container).toMatchSnapshot();
});

it("should call the filter function when user select the filter options", () => {
  const mockFunction = jest.fn();
  customRender(<Filter favArray={mockData} setallCardsArr={mockFunction} />);

  const filterButton = screen.getByText("FOOD & DRINK");
  userEvent.click(filterButton);

  expect(mockFunction).toBeCalled();
});
