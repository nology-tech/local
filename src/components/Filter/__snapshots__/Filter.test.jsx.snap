// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the filter component 1`] = `
<div>
  <div
    class="filter__list"
  >
    <button
      class="button button--filter"
    >
      FOOD & DRINK
    </button>
    <button
      class="button button--filter"
    >
      CULTURE
    </button>
    <button
      class="button button--filter"
    >
      OTHER
    </button>
  </div>
</div>
`;
